src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -451,6 +481,42 @@ public void SetSortingDirection(object sortingDir)
             this.Sort();
         }
 
+        /// <summary>
+        /// Set the sorting direction.  Used by the associated command.
+        /// </summary>
+        /// <param name="obj"></param>
+        public void ViewPackageDetails(object obj)
+        {
+            // to be implemented in future PR   
+        }
+
+        /// <summary>
+        /// Clears the search text box and resets the search.
+        /// </summary>
+        public void ClearSearchTextBox(object obj)
+        {
+            SearchText = "";
+        }
+
+        /// <summary>
+        /// When the user downloads new package via the package search manager, a toast notification
+        /// appears at the base of the window. This command fires when the user clicks to dismiss
+        /// one of these toast notifications.
+        /// </summary>
+        /// <param name="obj"></param>
+        public void ClearDownloadToastNotification(object obj)
+        {
+            if (!(obj is PackageDownloadHandle packageDownloadHandle)) return;
+
+            PackageDownloadHandle packageDownloadHandleToRemove = PackageManagerClientViewModel.Downloads
+                .FirstOrDefault(x => x.Id == packageDownloadHandle.Id);
```

Comment:
x is always valid ?

Reply:
I think so: if the button is visible for the user to click (and hence fire this method), then there should always be at least one item in the PackageManagerClientViewModel.Downloads collection.
We then check against null in the following statement.